rust/frontend/src/impls/service_based_frontend.rs

```diff
@@ -1577,6 +1578,140 @@ impl ServiceBasedFrontend {
         res
     }
 
+    pub async fn retryable_search(
+        &mut self,
+        request: SearchRequest,
+    ) -> Result<SearchResponse, QueryError> {
+        // TODO: The dispatch logic is mostly the same for count/get/query/search, we should consider unifying them
+        // Get collection and segments once for all queries
+        let collection_and_segments = self
+            .collections_with_segments_provider
+            .get_collection_with_segments(request.collection_id)
+            .await
+            .map_err(|err| QueryError::Other(Box::new(err) as Box<dyn ChromaError>))?;
+
+        let latest_collection_logical_size_bytes = collection_and_segments
+            .collection
+            .size_bytes_post_compaction;
+
+        // Aggregate metrics across all search payloads
+        let mut total_metadata_predicate_count = 0u64;
+        let mut total_fts_query_length = 0u64;
+        let mut total_search_embedding_count = 0u64;
+
+        for payload in &request.searches {
+            // Count metadata predicates and FTS query length from where clause
+            if let Some(ref where_clause) = payload.filter.where_clause {
+                total_metadata_predicate_count += where_clause.metadata_predicate_count();
+                total_fts_query_length += where_clause.fts_query_length();
+            }
+
+            // Count embeddings from the score expression
+            // Each rank in the score expression contains one embedding
+            let knn_queries = payload.rank.knn_queries();
+            total_search_embedding_count += knn_queries.len() as u64;
```

Comment:
We may not want to bill sparse vectors per dimension like this, we should discuss can we cut a ticket for this?